ml/rewrite/unification.py

```diff
@@ -37,6 +42,10 @@ def merge(self, other: UnificationResult) -> Optional[UnificationResult]:
 
         return UnificationResult(new_subst, self.applied_equations + other.applied_equations)
 
+    @staticmethod
+    def prepend_path_to_applied_eqs(applied_eqs: List[Tuple[Equation, PatternPath]], prefix: int) -> List[Tuple[Equation, PatternPath]]:
+        return [ (eqn, [ prefix ] + path) for eqn, path in applied_eqs ]
+
     def prepend_path(self, prefix: int) -> UnificationResult:
         return UnificationResult(self.substitution, [ (eqn, [ prefix ] + path) for eqn, path in self.applied_equations ])
```

Comment:
nit: since we have `prepend_path_to_applied_eqs` can we change this line to using `prepend_path_to_applied_eqs`?
(also can we rename it to `prepend_path_to_applied_equations`?)